File: scripts/utils/color_utils.py
class ColorUtils:
    """Utility class for applying ANSI color codes to text output."""
    RESET = '\033[0m'
    RED = '\033[31m'
    GREEN = '\033[32m'
    BOLD = "\033[1m"




    @staticmethod
    def info(text: str) -> str:
        
        return text
        
    @staticmethod
    def colorize(text: str, color: str, bold: bool = False) -> str:
        """Applies color and optional bold styling to the given text."""
        style = f"{ColorUtils.BOLD}" if bold else ""
        return f"{style}{color}{text}{ColorUtils.RESET}"

    @staticmethod
    def error(text: str) -> str:
        """Styles text as an error (red and bold)."""
        return ColorUtils.colorize(text, color=ColorUtils.RED, bold=True)

    @staticmethod
    def success(text: str) -> str:
        """Styles text as success (green and bold)."""
        return ColorUtils.colorize(text, color=ColorUtils.GREEN, bold=True)

    @staticmethod
    def warning(text: str) -> str:
        
        return text


def process_logs_and_highlight_errors(log: str) -> str:
    """Processes logs and highlights errors or warnings."""
    error_keywords = ["ERROR", "FAIL", "TimeoutError", "Traceback"]
    warning_keywords = ["WARN"]

    if any(keyword in log for keyword in error_keywords):
        return ColorUtils.error(log)
    if any(keyword in log for keyword in warning_keywords):
        return ColorUtils.warning(log)
    return log

File: core/tests/puppeteer-acceptance-tests/utilities/utils/color_utils.ts
export class ColorUtils {
    /** It is the Utility class for applying ANSI color codes to text output. */
    static RESET: string = '\u001b[0m';
    static RED: string = '\u001b[31m';
    static GREEN: string = '\u001b[32m';
   

    

    static error(text: string): string {
        /** It styles the entire block of text as red and bold. */
        return `\u001b[31m${text}\u001b[0m`;
    }
    

    static success(text: string): string {
        /** It styles text as success (green and bold). */
        return `${ColorUtils.GREEN}${text}${ColorUtils.RESET}`;
    }
    
}

export function processLogsAndHighlightErrors(logs: string): string {
    /** To wraps the entire log block in red ANSI codes. */
    return `\u001b[31m${logs}\u001b[0m`; // Entire block is wrapped in red
}

/**
 * To enhance console.error to ensure all error blocks are highlighted.
 */
export function enhancedConsoleError(logs: string): void {
    const processedLogs = processLogsAndHighlightErrors(logs);
    console.error(processedLogs);
}
